src/main/java/com/avispl/symphony/dal/infrastructure/management/neat/pulse/NeatPulseCommunicator.java

```diff
@@ -0,0 +1,1466 @@
+			});
+		}
+		return aggregatedDeviceList.stream().sorted(Comparator.comparing(item -> item.getProperties().get(DeviceInfo.ROOM_NAME.getPropertyName())))
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Adds or updates the aggregated device in the aggregated device list.
+	 *
+	 * @param aggregatedDevice The aggregated device to be added or updated.
+	 */
+	private void addOrUpdateAggregatedDevice(AggregatedDevice aggregatedDevice) {
+		boolean isExist = aggregatedDeviceList.stream().anyMatch(dev -> dev.getDeviceId().equals(aggregatedDevice.getDeviceId()));
+		if (isExist) {
+			aggregatedDeviceList.removeIf(dev -> dev.getDeviceId().equals(aggregatedDevice.getDeviceId()));
+		}
+		aggregatedDeviceList.add(aggregatedDevice);
+	}
+
+	/**
+	 * Populates monitor properties including device info, device sensor, device settings, and advanced controllable properties.
+	 *
+	 * @param cached The cached data containing device information.
+	 * @param stats The map to store monitor properties.
+	 * @param advancedControllableProperties The list to store advanced controllable properties.
+	 */
+	private void populateMonitorProperties(Map<String, String> cached, Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		populateDeviceInfo(cached, stats);
+		populateDeviceSensor(cached, stats);
+		populateDeviceSettings(cached, stats, advancedControllableProperties);
+	}
+
+	/**
+	 * Populates device information into the stats map.
+	 *
+	 * @param cached The cached data containing device information.
+	 * @param stats The map to store device information.
+	 */
+	private void populateDeviceInfo(Map<String, String> cached, Map<String, String> stats) {
+		for (DeviceInfo item : DeviceInfo.values()) {
+			String propertyName = item.getPropertyName();
+			String value = getDefaultValueForNullData(cached.get(propertyName));
+			switch (item) {
+				case MODEL:
+					break;
+				case CONNECTION_TIME:
+					stats.put(propertyName, convertDateTimeFormat(value));
+					break;
+				case FIRMWARE_UPDATE_VERSION:
+					String currentVersion = getDefaultValueForNullData(cached.get(DeviceInfo.FIRMWARE_CURRENT_VERSION.getPropertyName()));
+					if (NeatPulseConstant.NONE.equalsIgnoreCase(value)) {
+						stats.put(propertyName, value);
+					} else {
+						String updateAvailable = NeatPulseConstant.FALSE;
+						if (!value.equalsIgnoreCase(currentVersion)) {
+							stats.put(propertyName, value);
+							updateAvailable = NeatPulseConstant.TRUE;
+						}
+						stats.put("FirmwareUpdateAvailable", updateAvailable);
+					}
+					break;
+				case PRIMARY_MODE:
+					stats.put(propertyName, EnumTypeHandler.getValueByName(PrimaryModeEnum.class, value));
+					break;
+				case CONTROLLER_MODE:
+					stats.put(propertyName, EnumTypeHandler.getValueByName(ControllerModeEnum.class, value));
+					break;
+				case IN_CALL_STATUS:
+					stats.put(propertyName, EnumTypeHandler.getValueByName(CallStatusEnum.class, value));
```

Comment:
For callStatusEnum the possible values are too detailed for me - "On a teams call", "On a zoom call", may be a bit tricky to add triggers for these. I think we'll need to discuss this with Nicolas, if such values are allright :)